iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/consumer/SubscriptionConsumer.java

```diff
@@ -336,18 +336,23 @@ protected List<SubscriptionMessage> poll(final Set<String> topicNames, final lon
             case ERROR:
               final ErrorPayload payload = (ErrorPayload) pollResponse.getPayload();
               final String errorMessage = payload.getErrorMessage();
-              final boolean critical = payload.isCritical();
-              LOGGER.warn(
-                  "Error occurred when SubscriptionConsumer {} polling topics {}: {}, critical: {}",
-                  this,
-                  topicNames,
-                  errorMessage,
-                  critical);
-              if (critical) {
+              if (payload.isCritical()) {
                 throw new SubscriptionRuntimeCriticalException(errorMessage);
               } else {
                 throw new SubscriptionRuntimeNonCriticalException(errorMessage);
               }
+            case TERMINATION:
+              final SubscriptionCommitContext commitContext = pollResponse.getCommitContext();
+              final String topicNameToUnsubscribe = commitContext.getTopicName();
+              LOGGER.info(
+                  "Termination occurred when SubscriptionConsumer {} polling topics {}, unsubscribe topic {} on DN {} automatically",
+                  this,
+                  topicNames,
+                  topicNameToUnsubscribe,
+                  commitContext.getDataNodeId());
+              unsubscribe(topicNameToUnsubscribe);
```

Comment:
Note that since the pipe is automatically dropped, the "DropPipeProcedure"s in "DropSubscriptionProcedure" will encounter failure in validation... Need to add some checks or else the procedure will fail.

Reply:
In DropPipeProcedureV2, no exception will be occurred even if the pipe does not exist...

Reply:
Okay, checked that there are not.... However there will be a warn log in PipeDataNodeRemainingEventAndTimeMetrics#deregister. Can you remove it incidentally?

Reply:
> Okay, checked that there are not.... However there will be a warn log in PipeDataNodeRemainingEventAndTimeMetrics#deregister. Can you remove it incidentally?

added a boolean return value to `dropPipe`...